cms/server/AdminWebServer.py

```diff
@@ -523,6 +525,330 @@ def get(self, contest_id):
+            assert operation in (
+                "Remove",
+            ), "Please select a valid operation"
+        except Exception as error:
+            self.application.service.add_notification(
+                make_datetime(), "Invalid field(s)", repr(error))
+            self.redirect(fallback_page)
+            return
+
+        self.contest = self.safe_get_item(Contest, contest_id)
+
+        if operation == "Remove":
+            # Remove the participation.
+            participation = self.sql_session.query(Participation)\
+                .filter(Participation.user == user)\
+                .filter(Participation.contest == self.contest)\
+                .first()
+            self.sql_session.delete(participation)
+
+        if try_commit(self.sql_session, self):
+            # Create the user on RWS.
+            self.application.service.proxy_service.reinitialize()
+
+        # Maybe they'll want to do this again (for another contest).
+        self.redirect(fallback_page)
+
+
+class AssignContestTaskHandler(BaseHandler):
+    def post(self, contest_id):
+        fallback_page = "/contest/%s/tasks" % contest_id
+
+        self.contest = self.safe_get_item(Contest, contest_id)
+
+        try:
+            task_id = self.get_argument("task_id")
+            # Check that the admin selected some task.
+            assert task_id != "null", "Please select a valid task"
+        except Exception as error:
+            self.application.service.add_notification(
+                make_datetime(), "Invalid field(s)", repr(error))
+            self.redirect(fallback_page)
+            return
+
+        task = self.safe_get_item(Task, task_id)
+
+        # Assign the task to the contest.
+        task.num = len(self.contest.tasks)
+        task.contest = self.contest
+
+        if try_commit(self.sql_session, self):
+            # Create the user on RWS.
+            self.application.service.proxy_service.reinitialize()
+
+        # Maybe they'll want to do this again (for another task)
+        self.redirect(fallback_page)
+
+
+class ContestTasklistHandler(BaseHandler):
+    def get(self, contest_id):
+        self.contest = self.safe_get_item(Contest, contest_id)
+
+        self.r_params = self.render_params()
+        self.r_params["contest"] = self.contest
+        self.r_params["unassigned_tasks"] = \
+            self.sql_session.query(Task)\
+                .filter(Task.contest == None)\
+                .all()
+        self.render("contest_tasklist.html", **self.r_params)
+
+
+class EditContestTaskHandler(BaseHandler):
+    def post(self, contest_id):
+        fallback_page = "/contest/%s/tasks" % contest_id
+
+        self.contest = self.safe_get_item(Contest, contest_id)
+
+        try:
+            task_id = self.get_argument("task_id")
+            operation = self.get_argument("operation")
+            assert operation in (
+                "Remove from contest",
+                "Move up",
+                "Move down"
```

Comment:
As I understand the new AWS will have none of these things as it will be mainly a client side app. I'm not asking to share the constant between Python and JS, just give a name to these string in the class so you can use it both here and later.